### PR TITLE
Consolidated nav-bar components into App.js only.

### PR DESCRIPTION
**Pull request by secondaryfun**
- Related Issue (include '#'):  https://github.com/CotripperPlatform/CoTrip/issues/666 https://github.com/CotripperPlatform/CoTrip/issues/665

- Description of changes made:

1. Moved the nav-bar for each page into the respective Route on App.js
2. Added a couple placeholder functions to the Trip pages so the placeholder pages now render, allowing for testing of nav-bar render.
3. Removed nav-bars from all pages.
4. Modified Route on Member-Page to enable profile view lookups by prop.
5. Added profile image to correctly populate Nav-bar user image.

- Is the feature complete/bug resolved/etc..:  Yes

- Any known bugs/strange behavior:  Works cleanly, tested on all navigable pages currently on develop branch.

- Is there specific feedback you would like on these changes:  no

- Screenshot(s):
image
image
image

### CODE REVIEW

File: client/src/components/Layout/Layout.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+                path="/register"
+                exact
+                render={routerProps => (
+                    <OnboardingPage
+                        handleSignup={props.handleSignup}
+                        {...routerProps}
+                        logged_in={props.logged_in}
+                    />
+                )}
+            ></Route>
```
Comment (0xZakk) [inline client/src/components/Layout/Layout.js]: All the routing should stay in `App`, but be nested inside the `Layout` component. See #665 for what I mean

File: client/src/components/Layout/Layout.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    console.log(props)
+    return (
+        <div className="Layout">
+            {props.logged_in ? (
+                <Route
+                    path="/"
+                    exact
+                    render={routerProps => (
+                        <main>
+                            <Navbar to={`/member-page/${props.userid}`} profileImage={props.image} page={5}></Navbar>
```
Comment (0xZakk) [inline client/src/components/Layout/Layout.js]: We'll need some way to pass in the page number from App

**Review by 0xZakk — CHANGES_REQUESTED** [review]
@secondaryfun Looks good overall!

See my feedback

File: client/src/components/Layout/Layout.js (hunk 1 above)
Comment (secondaryfun) [reply client/src/components/Layout/Layout.js]: Refactored using props.children.

File: client/src/components/Layout/Layout.js (hunk 2 above)
Comment (secondaryfun) [reply client/src/components/Layout/Layout.js]: Page number determined by useLocation from react-router-dom.

File: client/src/components/Layout/Layout.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { useLocation } from "react-router-dom";
+
+import "./Layout.css";
+import Navbar from "../../components/Navbar/Navbar"
+import Footer from "../../components/Footer/Footer"
+
+
+export default function Layout(props) {
+    let currentPath = useLocation().pathname.split('/')[1]
+    console.log(currentPath)
```
Comment (0xZakk) [inline client/src/components/Layout/Layout.js]: Can you remove this and any other `console.log`s?

File: client/src/components/Layout/Layout.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    const pathPages = {
+        '': 99,
```
Comment (0xZakk) [inline client/src/components/Layout/Layout.js:13]: 🤔🤔🤔🤔🤔🤔

File: client/src/components/Layout/Layout.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        'forum-page-hashtag': 2,
+        'forum-page-discover': 2,
+        'forum-page-topic': 2,
+        'community': 1,
+        'directory': 0,
+    }
+
+    let page = pathPages[currentPath]
+    console.log(page)
+
```
Comment (0xZakk) [inline client/src/components/Layout/Layout.js]: Can you get rid of the variable and just do this on line 31?

File: client/src/components/Layout/Layout.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    return (
+        <main>
+            <Navbar to={`/member-page/${props.userid}`} profileImage={props.image} page={page}></Navbar>
+            {props.children}
+            <Footer history={props.history} handle_logout={props.handle_logout} />
+        </main>
```
Comment (0xZakk) [inline client/src/components/Layout/Layout.js]: This looks soooooo good!

**Review by 0xZakk — CHANGES_REQUESTED** [review]
@secondaryfun Just a few small changes - this looks really really good man! Really great work here!

File: client/src/components/Layout/Layout.js (hunk 4 above)
Comment (secondaryfun) [reply client/src/components/Layout/Layout.js:13]: The way the Navbar loads, if a page is out of range, then the triangle highlighter is not applied.  The previous code had the page set to 5, which is why you didn't notice it.  I set it to 99, so it is obvious which pages should not receive triangle.  This also allows for future nav menu items to be added without requiring code refactoring.

image

File: client/src/components/Layout/Layout.js (hunk 5 above)
Comment (secondaryfun) [reply client/src/components/Layout/Layout.js]: Done

File: client/src/components/Layout/Layout.js (hunk 6 above)
Comment (secondaryfun) [reply client/src/components/Layout/Layout.js]: :)

**Review by 0xZakk — APPROVED** [review]

**Pull request merged**